src/docfx/build/metadata/ContributionInfo.cs

```diff
@@ -190,15 +197,38 @@ private GitUserInfo ToGitUserInfo(UserProfile profile)
             {
                 var pathToDocset = group.Select(pair => pair.file.FilePath).ToList();
                 var pathToRepo = group.Select(pair => pair.pathToRepo).ToList();
-                var repoPath = group.Key.RepositoryPath;
-                var commitsList = GitUtility.GetCommits(repoPath, pathToRepo);
+                var repo = group.Key;
+                var repoPath = repo.RepositoryPath;
+                var (commitsByFile, allCommits) = GitUtility.GetCommits(repoPath, pathToRepo);
                 for (var i = 0; i < pathToDocset.Count; i++)
                 {
-                    result.Add(pathToDocset[i], commitsList[i]);
+                    result.Add(pathToDocset[i], commitsByFile[i]);
+                }
+                foreach (var commit in allCommits)
```

Comment:
consider running it in parallel.

Reply:
will consider this in #3190

---
In reply to: [211157360](https://github.com/dotnet/docfx/pull/3176#discussion_r211157360) [](ancestors = 211157360)